quic/s2n-quic-core/src/crypto/tls/offload.rs

```diff
@@ -37,10 +39,10 @@ impl<E: tls::Endpoint> OffloadEndpoint<E> {
 
         let handle = thread::spawn(move || {
```

Comment:
I don't think we want to spawn a new thread per session. This will essentially be an unbounded queue of spawning threads.